spartan/scripts/consolidate-sepolia-balances.sh

```diff
@@ -0,0 +1,98 @@
+# Usage: ./consolidate-sepolia-balances.sh <mnemonic> <funding_address> <values_file>
+# Environment variables:
+#   ETHEREUM_HOSTS (must be provided)
+
+# IMPORTANT NOTE: The script should be the same as the one found in the consolidate-balances.yaml template.
+# This standalone is left here for ad-hoc use if needed.
+
+set -exu
+
+mnemonic=$1
+funding_address=${2:-"0x33D525f5ac95c2BCf98b644738C7d5673480493A"}
+values_file=${3:-"ignition-testnet"}
+
+XDG_CONFIG_HOME=${XDG_CONFIG_HOME:-"$HOME/.config"}
+
+ETHEREUM_RPC_URL=$(echo "$ETHEREUM_HOSTS" | cut -d',' -f1)
+
+# Install cast if needed
+if ! command -v cast &>/dev/null; then
+  curl -L https://foundry.paradigm.xyz | bash
+  $HOME/.foundry/bin/foundryup && export PATH="$PATH:$HOME/.foundry/bin" ||
+    $XDG_CONFIG_HOME/.foundry/bin/foundryup && export PATH="$PATH:$XDG_CONFIG_HOME/.foundry/bin"
+fi
+
+# Get values from the values file
+value_yamls="../aztec-network/values/$values_file ../aztec-network/values.yaml"
+
+# Get the number of replicas for each service
+num_validators=$(./read_value.sh "validator.replicas" $value_yamls)
+num_provers=$(./read_value.sh "proverNode.replicas" $value_yamls)
+
+# Get the key index start values
+validator_key_index_start=$(./read_value.sh "aztec.validatorKeyIndexStart" $value_yamls)
+prover_key_index_start=$(./read_value.sh "aztec.proverKeyIndexStart" $value_yamls)
+bot_key_index_start=$(./read_value.sh "aztec.botKeyIndexStart" $value_yamls)
+
+# bots might be disabled
+bot_enabled=$(./read_value.sh "bot.enabled" $value_yamls)
+if [ "$bot_enabled" = "true" ]; then
+  num_bots=$(./read_value.sh "bot.replicas" $value_yamls)
+else
+  num_bots=0
+fi
+
+# Build an array of indices to check
+declare -a indices_to_check
+
+# Add validator indices
+for ((i = 0; i < num_validators; i++)); do
+  indices_to_check+=($((validator_key_index_start + i)))
+done
+
+# Add prover indices
+for ((i = 0; i < num_provers; i++)); do
+  indices_to_check+=($((prover_key_index_start + i)))
+done
+
+# Add bot indices if enabled
+if [ "$bot_enabled" = "true" ]; then
+  for ((i = 0; i < num_bots; i++)); do
+    indices_to_check+=($((bot_key_index_start + i)))
+  done
+fi
+
+echo "Checking balances for ${#indices_to_check[@]} accounts..."
+
+# For each index in our list
+for i in "${indices_to_check[@]}"; do
+  # Get address and private key for this index
+  address=$(cast wallet address --mnemonic "$mnemonic" --mnemonic-index $i)
+  private_key=$(cast wallet private-key --mnemonic "$mnemonic" --mnemonic-index $i)
+
+  # Get balance
+  balance=$(cast balance $address --rpc-url "$ETHEREUM_RPC_URL")
+
+  if [ "$balance" != "0" ]; then
+    gas_price=$(cast gas-price --rpc-url "$ETHEREUM_RPC_URL")
+    gas_price=$((gas_price * 120 / 100)) # Add 20% to gas price
+    gas_cost=$((21000 * gas_price))
+
+    # Calculate amount to send (balance - gas cost)
+    send_amount=$((balance - gas_cost))
```

Comment:
Shouldn't we also account for `priority_gas_price`?

Reply:
huh, hasn't been an issue when I've used this script, ig the 20% buffer that's added is enough?

Reply:
Maybe we're just sending legacy txs? Nvm then, if it works it works.